js/main.js

```diff
@@ -146,9 +147,47 @@ let checkbox_initializer = (thing, e) => {
     checkbox.addEventListener("change", cb);
 };
 
+let plusminus_initializer = (thing, e) => {
+    let div = e.querySelector("div.thing-detail");
+    let plus = e.querySelector("span[name='plus']");
+    let minus = e.querySelector("span[name='minus']");
+    let value_ele = e.querySelector("span[name='value']");
+    let new_value;
+    plus.addEventListener('click', () => {
+        //plus.click()
+        div.classList.add("pending");
+        new_value = parseInt(value_ele.innerText) + 1
+        console.log(new_value);
+        cb()
+    });
+    minus.addEventListener('click', () => {
+        //minus.click();
+        div.classList.add("pending");
+        new_value = parseInt(value_ele.innerText) - 1;
+        console.log(value_ele.innerText);
+        cb()
+    });
+
+    let cb = () => {
+
+        console.log(new_value);
+        socket.send(JSON.stringify({
+            type: "command",
+            id: thing.id,
+            value: new_value,
+        }));
+        add_pending_change(thing, setTimeout(() => {
+            value_ele.innerText = new_value;
```

Comment:
This would set the element to the new value regardless of success or not.

Reply:
yes. this behaviour is analogue to the behaviour of a button press. The button will change its colour if we press it and keep that colour when the confirmation comes from websocket.
Would you suggest other behaviour?

Reply:
The issue isn't with the pattern but the content of the function within the `setTimeout`. It will be called when the timeout for receiving an answer from the backend expires, which will currently set the text of the label to `new_value` (which is what we wanted to set it to if it had succeeded) instead of the previous value (which is still the current value as the server didn't respond in time when the timeout callback gets called).

Reply:
ok. how would you fix this?

Reply:
We can store `value_ele.innerText` in a variable before we set a the new value. The callback can use the stored value to restore `value_ele.innerText` if the `setTimeout` expires.